### DISCUSSION TITLE
Sign In With Otp - No Session after Email Link clicked

### DISCUSSION THREAD

**Discussion by merrick-vogt** [supabase/supabase#22379] [Questions]
Hi all, first time using supabase. I'm having an issue with my OTP login where the session is null after clicking "Log in" in the email. 

In Auth component:
	const handleLogin = async (e: FormEvent<HTMLFormElement>) => {
		e.preventDefault()
		try {
			setLoading(true)
			const { error } = await supabase.auth.signInWithOtp({ email })
			if(error) throw error
			alert("Check your email for the login link!")
		} catch (error){
			alert(error)
		} finally {
			setLoading(false)
		}
	}

In AuthSessionContext

	useEffect(() => {
		const auth = async () => {
			const {data, error } = await supabase.auth.getSession()

			console.log('session.auth', supabase.auth)
			console.log('session data', data);
			if(data.session){
				setSession(data.session)
				setLoading(false)
			} else {
				console.log('session error:', error)
			}
		}
		auth()
		supabase.auth.onAuthStateChange((_event, session) => {
			setSession(session)
			setLoading(false)
		})
	}, [])

The supabase.auth data shows in the logs, but the data from session is null. Why is a session not created? Or am I missing something else? Thanks in advance!

**Comment by andreicos1** [comment]
have you found a way to fix this?

**Reply by merrick-vogt** [reply]
I haven't yet. Any ideas?

**Reply by andreicos1** [reply]
I ended up using this and it seems to work https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr

Edit: This is a SSR flow so it won't work for you if you want to keep everything on the client unfortunately

**Reply by merrick-vogt** [reply]
Gotcha, thanks for the rec!
